test/metadata_edns0/metadata_edns0_test.go

```diff
@@ -0,0 +1,84 @@
+package metadataEdns0
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coredns/ci/test/kubernetes"
+	"github.com/coredns/coredns/plugin/test"
+	intTest "github.com/coredns/coredns/test"
+
+	"github.com/miekg/dns"
+)
+
+var tests = []test.Case{
+	{
+		Qname: "svc-1-a.test-1.svc.cluster.local.", Qtype: dns.TypeA,
+		Rcode: dns.RcodeSuccess,
+		Answer: []dns.RR{
+			test.A("svc-1-a.test-1.svc.cluster.local.      303    IN      A       10.96.0.100"),
+		},
+	},
+}
+
+func TestMetadata(t *testing.T) {
+	var f bytes.Buffer
+	log.SetOutput(&f)
+
+	corefileMeta := `.:0 {
+	   metadata
+       metadata_edns0 {
+          test 0xffee bytes
+       }
+       log . "Meta: {/metadata_edns0/test}"
+       forward . 8.8.8.8:53
+}
+`
+	meta, udpMeta, _, err := intTest.CoreDNSServerAndPorts(corefileMeta)
```

Comment:
That part seems odd.
=> you are running a DNS Server in your current thread.

I would think it would be a second Pod running in K8s .. no ?

I am not sure you need the CoreDNS with `kubernetes` plugin. What for ?
You would need:

1- DNS server that encode the EDNS0 and forward to ...
2- a second DNS server that decode as metadata and Log that metadata
3- collect the Log of that second server and check the content.

Unless we do not have access to the log of that second server ?